R/rdeephaven/src/client.cpp

```diff
@@ -0,0 +1,113 @@
+public:
+    TableHandleWrapper(deephaven::client::TableHandle ref_table) : internal_tbl_hdl(std::move(ref_table)) {};
+
+    TableHandleWrapper* select(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.select(columnSpecs));
+    };
+
+    TableHandleWrapper* view(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.view(columnSpecs));
+    };
+
+    TableHandleWrapper* dropColumns(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.dropColumns(columnSpecs));
+    };
+
+    TableHandleWrapper* update(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.update(columnSpecs));
+    };
+
+    SEXP getStream() {
+        ArrowArrayStream* stream = new ArrowArrayStream;
+
+        std::shared_ptr<arrow::flight::FlightStreamReader> fsr = internal_tbl_hdl.getFlightStreamReader();
+
+        // allocate memory for vector of RecordBatches and populate with fsr
+        std::vector<std::shared_ptr<arrow::RecordBatch>> empty_record_batches;
+        DEEPHAVEN_EXPR_MSG(fsr->ReadAll(&empty_record_batches)); // need to add OK or throw
+
+        // convert to RecordBatchReader
+        arrow::Result<std::shared_ptr<arrow::RecordBatchReader>> record_batch_reader = arrow::RecordBatchReader::Make(empty_record_batches);
+
+        //export to C struct
+        arrow::ExportRecordBatchReader(record_batch_reader.ValueOrDie(), stream);
+
+        // wrap pointer in external pointer and return
+        Rcpp::XPtr<ArrowArrayStream> xptr(stream, true);
+        return xptr;
+    };
+
+    void print() {
+        std::cout << internal_tbl_hdl.stream(true) << '\n';
+    };
+
+private:
+    deephaven::client::TableHandle internal_tbl_hdl;
+};
+
+
+class ClientWrapper {
+public:
+    TableHandleWrapper* emptyTable(int64_t size) {
+        return new TableHandleWrapper(internal_tbl_hdl_mngr.emptyTable(size));
+    };
+    TableHandleWrapper* openTable(std::string tableName) {
+        return new TableHandleWrapper(internal_tbl_hdl_mngr.fetchTable(tableName));
+    };
+    
+private:
+    ClientWrapper(deephaven::client::Client ref) : internal_client(std::move(ref)) {};
+    const deephaven::client::Client internal_client;
+    const deephaven::client::TableHandleManager internal_tbl_hdl_mngr = internal_client.getManager();
+    friend ClientWrapper* newClientWrapper(const std::string &target);
+};
+
+// factory method for calling private constructor, Rcpp does not like <const std::string &target> in constructor
+ClientWrapper* newClientWrapper(const std::string &target) {
+    return new ClientWrapper(deephaven::client::Client::connect(target));
+};
+
+
+// ######################### RCPP GLUE #########################
+
+using namespace Rcpp;
+
+RCPP_EXPOSED_CLASS(TableHandleWrapper)
+RCPP_EXPOSED_CLASS(BatchReader)
+RCPP_EXPOSED_CLASS(ArrowArrayStream)
+
+RCPP_MODULE(ClientModule) {
```

Comment:
Better module name.  Is `RDeephaven` or something else more appropriate?

Reply:
Depends on whether we break this file up into several, like `client.cpp`, `tableHandle.cpp`, etc. Then the Rcpp module at the bottom of each should be named appropriately.